src/rgw/rgw_op.cc

```diff
@@ -964,6 +965,24 @@ void rgw_build_iam_environment(rgw::sal::Driver* driver,
   }
 }
 
+void handle_replication_status_header(
+    const DoutPrefixProvider *dpp,
+    rgw::sal::Attrs& attrs,
+    rgw::sal::Driver* driver,
+    req_state* s,
+    ceph::real_time &obj_mtime) {
+  auto attr_iter = attrs.find(RGW_ATTR_OBJ_REPLICATION_STATUS);
+  if (attr_iter != attrs.end() && attr_iter->second.to_str() == "PENDING") {
+    RGWObjectSyncHandlerRef sync_handler;
+    int sync_ret = driver->get_object_sync_handler(dpp, &sync_handler, s->yield);
+    if (sync_ret == 0 && sync_handler->object_is_synced(dpp, s->object.get(), obj_mtime)) {
```

Comment:
maybe just add a `sal::Object::is_synced()` function that we can call directly instead of allocating/throwing away an 'object sync handler'?

maybe `is_pending_sync()` is a better name, because 'is synced' implies that sync was successful